### ISSUE TITLE
how to

### ISSUE THREAD

**Issue by Meliordass** [hasantha-nirmal/speed_violation_detection_yolo#1]
How to change the model for yolov4?I try to put my pb file in checkpoint,but python give me an error.

**Comment by hasantha-nirmal** [comment]
Hi, Did you execute the following command after saving the yolov4.weights to /data folder?
python save_model.py --model yolov4
This is where the yolo weights are transferred to tf weights. Don't directly put the .pb file to any location. This command will take care of that.

**Comment by Meliordass** [comment]
Thank you very much for your reply, I may not have formulated my question well. I know the method you said, yolov4.weights is the pre-trained weights of yolov4, but if only this file is used to generate saved_model.pb, the prediction result for the vehicle is very poor, so I want to use another code to train the model for the vehicle first to improve Accuracy, but using my trained model as an object detector for your code seems to have a problem and I don't know what to do. Very eager for your guidance! Thank you!

**Comment by hasantha-nirmal** [comment]
Hi, I understand that you need to do the object tracking with your own custom_yolov4.weights. The current repo has set to read the default yolov4 weights. To run your custom yolov4 model, first copy your custom_yolov4.weights to the data folder. Make sure to rename your custom_yolov4 weights to "yolov4.weights" since, that is the name that will read by the save_model.py. Then go to the data/classes folder and edit the "coco.names" with the list that you used to train your custom model. (coco.names file need to be as same as the file you used to train your yolov4.weights, simply replace it with the exiting one) Afterwards, runs the program as instructed on the repo. Make sure to edit the following lines of speed, laneline and parking viol.py files. 

#by default allow all classes in .names file
#allowed_classes = list(class_names.values())

currently I have restricted only several classes such as bus and trucks.

Moreover, if you could paste the error next time, that would be more helpful. Even a snapshot would be fine. 

Hope this helps :)

**Comment by Meliordass** [comment]
Hi, thank you very much for your guidance, it worked very well. I'm very happy to tell you: I used darknet to train my own dataset in the past few days and got a weights file, I put it in the data folder and replaced coco.data with the .names file I made (replace coco .names file happens to be something I overlooked earlier). Finally, I used save_model.py to convert the weight file into a pb file, and it ran successfully and got a good recognition result. thank you very much  :)
